packages/modules/vehicles/mercedeseq/api.py

```diff
@@ -0,0 +1,209 @@
+soc_url_pre_vin = "https://api.mercedes-benz.com/vehicledata/v2/vehicles/"
+soc_url_post_vin = "/containers/electricvehicle"
+
+log = logging.getLogger("soc."+__name__)
+
+soc = None
+range = None
+timestamp = None
+
+
+def socDebugLog(message):
+    log.debug(": " + message)
+
+
+def handleResponse(what, status_code, text):
+    if status_code == 204:
+        # this is not an error code. Nothing to fetch so nothing to update
+        socDebugLog(what + " Request Code: " + str(status_code) +
+                    " (no data is available for the resource)")
+        socDebugLog(text)
+    elif status_code == 400:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (Bad Request)")
+        socDebugLog(text)
+
+    elif status_code == 401:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (Invalid or missing authorization in header)")
+        socDebugLog(text)
+
+    elif status_code == 402:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (Payment required)")
+        socDebugLog(text)
+
+    elif status_code == 403:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (Forbidden)")
+        socDebugLog(text)
+
+    elif status_code == 404:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (The requested resource was not found, e.g.: the selected vehicle could not be found)")
+        socDebugLog(text)
+
+    elif status_code == 429:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (The service received too many requests in a given amount of time)")
+        socDebugLog(text)
+
+    elif status_code == 500:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (The service received too many requests in a given amount of time)")
+        socDebugLog(text)
+
+    elif status_code == 503:
+        socDebugLog(what + " Request fehlgeschlagen Code: " + str(status_code) +
+                    " (The server is unable to service the request due to a temporary unavailability condition)")
+        socDebugLog(text)
+
+    else:
+        socDebugLog(what + " Request fehlgeschlagen unbekannter Code: " + str(status_code))
+        socDebugLog(text)
+
+
+def fetch_soc(config: MercedesEQSoc,
+              vehicle_id: int) -> Tuple[float, float]:
+    if Debug >= 1:
+        socDebugLog("Debug Level: " + str(Debug))
+
+    client_id = config.configuration.client_id
+    client_secret = config.configuration.client_secret
+    vin = config.configuration.vin
+    soc_url = soc_url_pre_vin + str(vin) + soc_url_post_vin
+    vehicle = vehicle_id
+    if Debug >= 1:
+        socDebugLog("client: " + client_id)
+
+    if Debug >= 2:
+        socDebugLog("SOC URL: " + soc_url)
```

Comment:
```suggestion
    log.info("client: " + client_id)
    log.debug("SOC URL: " + soc_url)
```
Mit dem Python-Logging lässt sich das so zusammenfassen.

Reply:
done